src/Symfony/Component/Validator/Constraints/StringContains.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+/**
+ * @Annotation
+ * @Target({"PROPERTY", "METHOD", "ANNOTATION"})
+ */
+class StringContains extends AbstractStringContains
+{
+    public const NOT_CONTAINS_ERROR = '2e0f55ec-eb23-411d-8480-303ab84fb389';
+
+    protected static $errorNames = [
+        self::NOT_CONTAINS_ERROR => 'NOT_CONTAINS_ERROR',
+    ];
+
+    /**
+     * @var string
+     */
+    public $message = 'This does not contain expected value.';
```

Comment:
Maybe we can provide the expected text in the message and change the it to something like :

`This value does not contain the expected text: "{{ text }}".`

Reply:
There are two reasons why it doesn't expose `text`/`expected` yet:

- `text` accepts an array, a long list may make it unreadable
- `text` may contains sensible/security data

Anyway, I'll try to use your suggestion if `text` has a single item.